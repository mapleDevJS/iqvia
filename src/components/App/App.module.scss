.app {
    display: flex;
    align-items: center;
    height: 100vh;

    @media (max-width: 70rem) {
        display: block;
    }
}

.container {
    display: grid;
    width: 100%;
    min-width: var(--viewport--sm);
    max-width: var(--viewport--xl);
    margin: 0 auto;
    padding: 0 2rem;
    grid-template-columns: 32rem minmax(32rem, 1fr);
    gap: 2rem;

    @media (max-width: 70rem) {
        max-width: var(--viewport--md);
        padding: 1rem;
        grid-template-columns: auto;
    }
}
