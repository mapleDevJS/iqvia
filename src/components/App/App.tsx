import { observer } from 'mobx-react';
import { useContext } from 'react';
import styles from './App.module.scss';
import utilityStyles from 'styles/utils.module.scss';
import ForecastDetails from 'components/Forecast/Forecast';
import Locations from 'components/Locations/Locations';
import { store } from 'store/index';

function App() {
    const { locationsStore } = useContext(store);
    const { getForecast } = locationsStore;

    const forecast = getForecast();

    return (
        <div className={styles.app}>
            <main className={styles.container}>
                <h1 className={utilityStyles.visuallyHidden} data-testid="visibility-hidden">World Weather Forecast</h1>
                <Locations />
                {forecast && <ForecastDetails forecast={forecast} />}
            </main>
        </div>
    );
}

export default observer(App);
