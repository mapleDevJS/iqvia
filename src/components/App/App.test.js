import { render, screen } from '@testing-library/react';
import App from './App';
// import ForecastDetails from 'components/Forecast/Forecast';

jest.mock('../Forecast/Forecast.tsx', () => () => <div data-testid='forecast'></div>);


describe('when rendered', () => {
    test('renders App component', () => {
        render(<App />);
    });
    it('should contain an expected title', () => {
        render(<App />);
        const titleElement = screen.getByText(/weather/i);
        expect(titleElement).toBeInTheDocument();
    });

    it('should visually hide an title', () => {
        render(<App />);
        const titleElement = screen.getByText(/World Weather Forecast/i);
        expect(titleElement).toHaveClass('visuallyHidden', { exact: true });
    });

    it('should contain an main section', () => {
        render(<App />);
        const mainElement = screen.getByRole('main');
        expect(mainElement).toBeInTheDocument();
    });

    it('should not render ForForecastDetails component',() => {
        render(<App />);
        expect(screen.queryByTestId('forecast')).toBeNull();
    });
});

describe('snapshot', () => {
    it('should take a snapshot', () => {
        const { asFragment } = render(<App />);

        expect(asFragment(<App />)).toMatchSnapshot();
    });
});
