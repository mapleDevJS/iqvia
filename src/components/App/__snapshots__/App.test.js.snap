// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <main
      class="container"
    >
      <h1
        class="visuallyHidden"
        data-testid="visibility-hidden"
      >
        World Weather Forecast
      </h1>
      <section
        class="locations"
      >
        <div
          class="searchBox"
        >
          <input
            class="inputLocation"
            placeholder="Type city name"
            value=""
          />
          <button
            class="button undefined"
          >
            Add
          </button>
        </div>
        <h2
          class="title"
        >
          Recent Locations
        </h2>
        <div
          class="bottomControls"
        >
          <button
            class="button undefined"
          >
            Clear
          </button>
        </div>
      </section>
    </main>
  </div>
</DocumentFragment>
`;
